utils.go

```diff
@@ -81,3 +83,120 @@ func toNRGBA(img image.Image) *image.NRGBA {
 	}
 	return Clone(img)
 }
+
+// nrgbaToHSL converts NRGBA to HSL.
+func nrgbaToHSL(c color.NRGBA) (float64, float64, float64) {
+	var h, s, l float64
+
+	r := float64(c.R) / float64(255)
+	g := float64(c.G) / float64(255)
+	b := float64(c.B) / float64(255)
+
+	min := math.Min(math.Min(r, g), b)
+	max := math.Max(math.Max(r, g), b)
+
+	l = (max + min) / 2
+
+	if min == max {
+		s = 0
+		h = 0
+	} else {
+		if l < 0.5 {
+			s = (max - min) / (max + min)
+		} else {
+			s = (max - min) / (2.0 - max - min)
+		}
+
+		if max == r {
+			h = (g - b) / (max - min)
+		} else if max == g {
+			h = 2.0 + (b-r)/(max-min)
+		} else {
+			h = 4.0 + (r-g)/(max-min)
+		}
+
+		h *= 60
+
+		if h < 0 {
+			h += 360
+		}
+	}
+
+	return h, s, l
+}
+
+// hslToNRGBA converts HSL to NRGBA with A=1.
+func hslToNRGBA(h, s, l float64) color.NRGBA {
+	if s == 0 {
+		c := uint8(l * 255)
```

Comment:
I think rounding the value will give more accurate result. E.g. `clamp(l * 255)` instead of `uint8(l * 255)`